front/src/common/RollingStockSelector/RollingStock2Img.tsx

```diff
@@ -1,44 +1,38 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import nextId from 'react-id-generator';
-import PropTypes from 'prop-types';
 import { LazyLoadImage } from 'react-lazy-load-image-component';
+import { RollingStock } from 'common/api/osrdEditoastApi';
 import { get } from '../requests';
 
-function RollingStock2Img(props) {
-  const { rollingStock } = props;
+const RollingStock2Img: React.FC<{ rollingStock: RollingStock }> = ({ rollingStock }) => {
   // as the image is stored in the database and can be fetched only through api (authentication needed),
   // the direct url can not be given to the <img /> directly. Thus the image is fetched, and a new
   // url is generated and stored in imageUrl (then used in <img />).
-  const [imageUrl, setImageUrl] = useState(null);
+  const [imageUrl, setImageUrl] = useState('');
 
-  const getRollingStockImage = useCallback(async () => {
-    if (!rollingStock || !Array.isArray(rollingStock.liveries)) return;
+  const getRollingStockImage = async () => {
+    const { id, liveries } = rollingStock;
+    if (!rollingStock || !Array.isArray(liveries)) return;
 
-    const defaultLivery = rollingStock.liveries.find((livery) => livery.name === 'default');
+    const defaultLivery = liveries.find((livery) => livery.name === 'default');
     if (!defaultLivery?.id) return;
 
     try {
-      const image = await get(`/rolling_stock/${rollingStock.id}/livery/${defaultLivery.id}/`, {
+      const image = await get(`/editoast/rolling_stock/${id}/livery/${defaultLivery.id}/`, {
         responseType: 'blob',
       });
       if (image) setImageUrl(URL.createObjectURL(image));
     } catch (e) {
-      /* empty */
+      console.error(e);
     }
-  }, [rollingStock]);
+  };
 
   useEffect(() => {
-    setImageUrl(null);
+    setImageUrl('');
     getRollingStockImage();
-  }, [getRollingStockImage]);
+  }, [rollingStock]);
 
   return imageUrl ? <LazyLoadImage src={imageUrl} alt={rollingStock?.name} key={nextId()} /> : null;
```

Comment:
Please use something else than `nextId()`, why not create a unique id key with a prefix associated to rolling stock unique name ?